File: frontend/src/interfaces/jobApplication.ts
// User ของ Student
export interface User {
  ID: number;
  username: string;
}

// ข้อมูลธนาคาร
export interface Bank {
  bank_name: string;
}

// ข้อมูลบริษัท
export interface Employer {
  company_name: string;
  logo?: string; // optional, ถ้ามีโลโก้
}

// ข้อมูลประกาศงาน
export interface JobPost {
  ID: number;
  title: string;
  Employer: Employer;
  image_url?: string;
}

//  เพิ่ม interface InterviewScheduling
export interface InterviewScheduling {
  ID: number;
  DateAndTime: string;  // backend ส่ง ISO string เช่น "2025-09-08T10:00:00Z"
  Status: string;
  Description?: string;
}

export type ApplicationStatus =
  | "Pending"
  | "InterviewPending"
  | "InterviewScheduled"
  | "Interviewed"
  | "Accepted"
  | "Rejected"
  | "Cancelled";

//  ใช้ Student ตัวจริงจาก interfaces/student.ts
import type { Student } from "./student";

// ข้อมูลการสมัครงาน
export interface JobApplication {
  ID: number;
  application_status: ApplicationStatus;   
  application_reason: string;
  CreatedAt: string;
  JobPost: JobPost;
  Student: Student;   // ตอนนี้ Student มี profile_image_url แน่นอน 
  resume_file?: string; //  เพิ่มฟิลด์นี้
  InterviewScheduling?: InterviewScheduling | null;
}

File: backend/entity/jobapplication.go
package entity

import (
	"time"
	"gorm.io/gorm"
)

type ApplicationStatusEnum string

const (
	StatusPending            ApplicationStatusEnum = "Pending"            // รอพิจารณา
	StatusInterviewPending   ApplicationStatusEnum = "InterviewPending"   // รอเลือกวันสัมภาษณ์
	StatusInterviewScheduled ApplicationStatusEnum = "InterviewScheduled" // เลือกวันสัมภาษณ์แล้ว
	StatusInterviewed        ApplicationStatusEnum = "Interviewed"        // สัมภาษณ์เสร็จแล้ว
	StatusAccepted           ApplicationStatusEnum = "Accepted"           // ผ่านการคัดเลือก
	StatusRejected           ApplicationStatusEnum = "Rejected"           // ไม่ผ่านการคัดเลือก
    StatusCancelled          ApplicationStatusEnum = "Cancelled"          // ยกเลิกการสมัคร เพิ่มอันนี้
)

type JobApplication struct {
    gorm.Model

    ApplicationStatus  ApplicationStatusEnum `gorm:"type:varchar(50);not null" json:"application_status"`
    LastUpdate        time.Time              `gorm:"not null" json:"last_update"`
    ApplicationReason string                 `gorm:"type:varchar(255);not null" json:"application_reason"`

    //บอกตัวเอง อย่าลืมไปเพื่อมตาราง JobApplication (1) to (1) Interview
   // FK Interview (1:1) ขแยาดเปลี่ยน ชั้นคิดผิดTTTT
   

    Interview   *Interview `gorm:"foreignKey:JobApplicationID" json:"interview"`

	//FK

    StudentID uint      `gorm:"not null" json:"student_id"`
    Student   Student `gorm:"foreignKey:StudentID;references:ID"`  // FK to Student.ID

    JobPostID uint    `gorm:"not null" json:"job_post_id"`
    JobPost   Jobpost `gorm:"foreignKey:JobPostID;references:ID"`  // FK to JobPost.ID

    //บอกตัวเอง อย่าลืมไปเพื่อมตาราง JobApplication (1) to (1) InterviewScheduling
    // เพิ่ม FK เพื่อเชื่อมกับตาราง InterviewScheduling
	// InterviewSchedulingID *uint               `json:"interview_scheduling_id"`
	// InterviewScheduling   *InterviewScheduling `gorm:"foreignKey:InterviewSchedulingID" json:"interview_scheduling"`

     // เพิ่มช่องเก็บไฟล์ ResumeFile
    ResumeFile string `gorm:"type:varchar(255)" json:"resume_file"`
    
}